server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -660,12 +683,22 @@ export class AssetService {
         if ((await this._albumRepository.getSharedWithUserAlbumCount(authUser.id, assetId)) > 0) {
           continue;
         }
-
-        //TODO: Step 3: Check if asset is part of a public album
       }
       throw new ForbiddenException();
     }
   }
+
+  async checkUploadAccess(sharedKey: string, assetPath: string) {
+    const sharedLink = await this.shareCore.getSharedLinkByKey(sharedKey);
+    if (!sharedLink.allowUpload) {
+      await this.backgroundTaskService.deleteFileOnDisk([
+        {
+          originalPath: assetPath,
+        } as any,
+      ]); // s
+      throw new ForbiddenException();
+    }
+  }
```

Comment:
We should prevent the asset from being uploaded entirely if they don't have upload access (do it in the file filter).